### PR TITLE
some updates on Chapter Basic of the documentaion

### PR DESCRIPTION
**Pull request by WeiqunZhang**

### CODE REVIEW

File: Docs/sphinx_documentation/source/Basics.rst [hunk 1]
```diff
@@ -49,6 +56,17 @@ single precision. :cpp:`amrex::Real` is typedef’d to either :cpp:`double` or
 
         use amrex_fort_module, only : amrex_real
 
+In C++, AMReX also provides a user literal :cpp:`_rt` so that one can
+have a proper type for constants (e.g., :cpp:`2.7_rt`).
+
+Long
+====
+
+AMReX defines a 64 bits integer type :cpp:`amrex::Long` that is an alias to
```
Comment (atmyers) [inline Docs/sphinx_documentation/source/Basics.rst:65]: `64 bit`

File: Docs/sphinx_documentation/source/Basics.rst [hunk 2]
```diff
@@ -1067,6 +1093,22 @@ FArrayBox.
       Real a = 3.0;
       fab2.saxpy(a, fab1); // For both components, fab2 <- a * fab1 + fab2
 
+These floating point operation functions are templated with parameter
+:cpp:`RunOn` specifying where they run, :cpp:`RunOn::Host` or
+:cpp:`RunOn::Device`.  When AMReX is built just for CPU, the
+template parameter has a default value of :cpp:`RunOn::Host` so that
+the user does not need to specify it for backward compatibility
+reason, and if :cpp:`RunOn::Device` is provided it will be ignored.
```
Comment (atmyers) [inline Docs/sphinx_documentation/source/Basics.rst:1101]: "for backward compatibility,"

Comment (WeiqunZhang) [reply Docs/sphinx_documentation/source/Basics.rst:1101]: Thanks for proofreading!

File: Docs/sphinx_documentation/source/Basics.rst [hunk 3]
```diff
@@ -2110,39 +2152,56 @@ These loops should usually use :cpp:`i <= hi.x`, not :cpp:`i < hi.x`, when
...
+      Array4<Real> const& a = mfa[mfi].array();
+      Array4<Real const> const& b = mfb[mfi].const_array();
+      Array4<Real const> const& c = mfc[mfi].const_array();
+      ParallelFor(bx, [=] AMREX_GPU_DEVICE (int i, int j, int k)
+      {
+        a(i,j,k) += b(i,j,k) * c(i,j,k);
+      });
+    }
+
+Here, :cpp:`PareallelFor` takes two arguments.  The first argument is
```
Comment (atmyers) [inline Docs/sphinx_documentation/source/Basics.rst:2181]: :cpp:`ParallelFor`

File: Docs/sphinx_documentation/source/Basics.rst [hunk 4]
```diff
@@ -2110,39 +2152,56 @@ These loops should usually use :cpp:`i <= hi.x`, not :cpp:`i < hi.x`, when
...
-Note that, unlike :cpp:`ParalleFor` that will be introduced in section
-:ref:`sec:gpu:for`, they do not launch GPU kernels to do the work,
-although they can be used in GPU kernels.
+The code above is performance portable.  It works with and without
+GPU.  When AMReX is built with GPU support, AMREX_GPU_DEVICE indicates
+that the lambda function is a device function and :cpp:`ParallelFor`
+launches a GPU kernel to do the work.  When it is built without GPU
+support, AMREX_GPU_DEVICE has no effects whatsoever.  More details on
+:cpp:`ParalleFor` that will be presented in section
+:ref:`sec:gpu:for`.
```
Comment (atmyers) [inline Docs/sphinx_documentation/source/Basics.rst:2204]: Based on this discussion, I think some people will get the idea that `ParallelFor` does OpenMP. Maybe we should add a note that this is not the case, and that OpenMP is meant to handled in the outer `MFIter` loops.

Comment (WeiqunZhang) [reply Docs/sphinx_documentation/source/Basics.rst:2204]: sure

Comment (WeiqunZhang) [reply Docs/sphinx_documentation/source/Basics.rst]: What do you think about the current version?  Please feel free to edit directly.

**Pull request merged**